tests/phpunit/Unit/SQLStore/QueryDependency/QueryDependencyLinksStoreTest.php

```
@@ -753,6 +755,225 @@ public function testdoUpdateDependenciesByFromQueryResult() {
+		// would short-circuit the second invocation entirely, leaving only
+		// two `add` calls and the original "first wins" data-loss shape.
+		// This assertion is the regression sentinel against the dedup
+		// re-appearing at this layer.
+		$this->assertCount( 4, $addIndices,
+			'Both invocations must reach addToUpdateList twice each; got: ' . json_encode( $callOrder ) );
+		$this->assertLessThan( $flushIndex, max( $addIndices ),
+			'Every addToUpdateList call must precede the first doUpdate flush so dep lists accumulate in $updateList before the DELETE-then-INSERT write; got: ' . json_encode( $callOrder ) );
+
+		// At most one `flush` should run for a single-request batch (one or
+		// more drains may be queued but only the first does real work).
+		$flushCount = count( array_keys( $callOrder, 'flush', true ) );
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$flushCount'. (undefined)

(UnusedLocalVariable)